### DISCUSSION TITLE
Repository titles on Google search results

### DISCUSSION THREAD

**Discussion by Scotchester** [community/community#16110] [Other Feature Feedback, Questions, & Ideas]
**tl;dr:** Could GitHub work with Google to ensure that search results for repositories have consistent titles?

---

Occasionally when I am in a rush and not paying enough attention, I'll put something in my address bar that I expect will use my history or bookmarks to take me to a GitHub repo, but it triggers a Google search instead.

I was recently dismayed to see Google using this title on the result for a project I contribute to, [wagtail/wagtail](https://github.com/wagtail/wagtail):

<img width="908" alt="Google search for 'wagtail/wagtail' with a result for the GitHub repo showing a title of 'wagtail.org - GitHub'" src="https://user-images.githubusercontent.com/1044670/167020519-d6afa8d8-1009-4d3c-b33a-df1b2a841e5d.png">

Repository results used to use the `<title>` element directly, which is sensibly formatted by GitHub as `<title>user-or-org/repo-name: repository description</title>`, then appending ` - GitHub` to it. From testing searches for some other repos I use, seems like they still do in many cases. But at least sometimes, it appears [they're trying to use some of their infinite wisdom to determine what a more helpful result title would be](https://developers.google.com/search/blog/2021/08/update-to-generating-page-titles).

In the example I gave above "wagtail.org" is a terrible choice for the title of the wagtail/wagtail result, because it loses the `owner/repo` convention we all know so well, _and_ there is a totally separate [wagtail/wagtail.org repo](https://github.com/wagtail/wagtail.org) for the project website!

I guess Google is deriving their custom title from the link in the sidebar here?

<img width="361" alt="wagtail/wagtail repo sidebar with wagtail.org link in it" src="https://user-images.githubusercontent.com/1044670/167023106-adeb30a1-0805-4581-bf99-9fbf46eb08f0.png">

I don't know if there is anything GitHub itself can actually do about this, but maybe it could at least use its influence to lean on Google to tweak their results for GitHub repo homepages? [This worked successfully for the team that works on the UK NHS website.](https://torchbox.com/blog/influencing-google-page-title-update/)

**Comment by thibaudcolas** [comment]
Trying out a few other cases outside Wagtail, this seems like a very common issue. I got good results looking for ESLint repositories, and here are a few Microsoft ones to illustrate:

![github-titles-serp](https://user-images.githubusercontent.com/877585/167030428-73811f47-54dc-409d-b06b-968b4d0a6a6e.png)

Those aren’t nearly as bad, but in my opinion they nonetheless are problematic because they make it harder to identify which page is a  repository (vs. a wiki page, issue, etc.)
